src/test/java/com/mongodb/jdbc/MongoDriverTest.java

```diff
@@ -200,4 +202,32 @@ void testGetPropertyInfo() throws SQLException {
         assertEquals(res.length, 1);
         assertEquals(res[0].name, "user");
     }
+
+    @Test
+    void testDialectProperty() throws SQLException {
+        MongoDriver d = new MongoDriver();
+        Properties p = new Properties();
+        Connection c;
+
+        // dialect set to "mysql" results in MySQLConnection
+        p.setProperty("dialect", "MySQL");
```

Comment:
Let's add one more test without this property set to confirm the default value of MySQL

Reply:
oh right! I had that in mind and then forgot to actually add it. The other tests are technically all covering that case since none of them set it, but I'll ensure here that without it set we return an `instanceof MySQLConnection`.